### PR TITLE
Add endpoint to get model run options

### PR DESCRIPTION
**Pull request by r-ash**
This PR will

* Add an endpoint to extract model run options from JSON template in naomi and input data
* Moves some of the data reading into a common location (thinking we can use this when we come to add a caching layer in the future)
* Update docker build to download latest version of hintr from github

This PR is getting quite big so I will do the work to return the full model options in a separate PR after this one. That should involve just updating the template in naomi and making more data available in params. See https://vimc.myjetbrains.com/youtrack/issue/mrc-574 for changes needed in next iteration of this.

### CODE REVIEW

File: R/model_options.R [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+#' @param anc Path to optional ANC file
+#'
+#' @return The model options declarative JSON UI.
+#' @keywords internal
+do_endpoint_model_options <- function(options_template, shape, survey,
+                                      programme, anc) {
+  regions <- read_geojson_regions(shape)
+  parent_region <- regions[!grepl("\\.", regions)]
+  if (length(parent_region) != 1) {
+    stop(sprintf("Should have located one parent regions but found regions %s.",
```
Comment (richfitz) [inline R/model_options.R]: if this is a likely problem, should you be checking that all regions have the same start? (`length(unique(sub("\\..+$", "", regions))) == 1`)

File: R/model_options.R [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+      e$message <- paste0(
+        "Failed to construct model options from template and params:\n",
+        e$message)
+      stop(e)
+    }
+  )
+}
+
+collapse_and_quote_transformer <- function(text, envir) {
+  res <- eval(parse(text = text, keep.source = FALSE), envir)
```
[the remark this answers is not in the document]
Comment (r-ash) [reply R/model_options.R]: Yep it will be - I've updated that

File: R/endpoints.R [hunk 3]
```diff
@@ -1,11 +1,13 @@
-api_build <- function(queue) {
+api_build <- function(queue, options_template) {
```
[the remark this answers is not in the document]
Comment (r-ash) [reply R/endpoints.R]: No need - moved that.

File: R/model_options.R [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+  art_quarter_options <- NULL
+  if (!is.null(programme)) {
+    art_quarter_options <-
+      naomi::quarter_year_labels(read_quarters(programme))
+  }
+  ## We will need these when we move to full spec of UI - leaving in for now
+  survey_options <- read_surveys(survey)
+  if (!is.null(anc)) {
+    anc_quarter_options <- naomi::quarter_year_labels(read_quarters(anc))
+    ## TODO: write these into appropriate part of naomi template when avaialble
```
[the remark this answers is not in the document]
Comment (r-ash) [reply R/model_options.R]: I've clarified that line and added a link to the task

File: R/model_options.R [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+#' @param survey Path to survey file
+#' @param programme Path to optional programme file
+#' @param anc Path to optional ANC file
+#'
+#' @return The model options declarative JSON UI.
+#' @keywords internal
+do_endpoint_model_options <- function(options_template, shape, survey,
+                                      programme, anc) {
+  regions <- read_geojson_regions(shape)
+  parent_region <- regions[!grepl("\\.", regions)]
```
[the remark this answers is not in the document]
Comment (r-ash) [reply R/model_options.R]: Thanks!

**Pull request merged**